### PR TITLE
Use default provider credentials when none are specified

### PR DESCRIPTION
**Pull request by negz**
Addresses https://github.com/crossplane/example-cnp/issues/2 and https://github.com/crossplane/example-cnp/issues/3.

### CODE REVIEW

File: eks.clusters.cnp.example.org/composition.yaml [hunk 1]
```diff
@@ -405,8 +367,15 @@ spec:
...
+      # ProviderCredentials is synonymous with the contemporary Provider kind.
+      # This sketch proposes it be renamed to avoid confusion with the newly
+      # introduced Provider kind (in the pkg.crossplane.io group) and to better
+      # clarify its purpose as a pointer to credentials. Note also that this
+      # type cannot currently be used in a composition because it is lacking the
+      # status conditions that a composite resource depends on to determine
+      # readiness.
       apiVersion: kubernetes.crossplane.io/v1alpha1
-      kind: Provider
+      kind: ProviderCredentials
```
Comment (lukeweber) [inline eks.clusters.cnp.example.org/composition.yaml]: Maybe this will be AWSProviderCredentials as we dedupe the types.

**Review by lukeweber — APPROVED** [review]

File: eks.clusters.cnp.example.org/composition.yaml (hunk 1 above)
Comment (negz) [reply eks.clusters.cnp.example.org/composition.yaml]: It sounds like we're going for `ProviderConfig` per our discussion today. I'll address that in another PR. Whether we prefix it with `AWS` (or similar) is a good question. Technically the API group already differentiates what provider its for (e.g. aws.crossplane.io vs gcp.crossplane.io) but that does fall prey to the kubectl-only-lists-the-lexicographically-first-API-group quirk in that `kubectl get providerconfig` would only ever show you Alibaba providers (until someone installed an Aardvark provider).

Comment (negz) [reply eks.clusters.cnp.example.org/composition.yaml]: Ah I guess I was originally addressing the name in this PR too. I'll update to `ProviderConfig` for now.

**Review by prasek — APPROVED** [review]
LGTM, thx!

**Pull request merged**